src/Formats/ProtobufHDFSSourceTree.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#if !defined(ARCADIA_BUILD)
+#    include <Common/config.h>
+#    include "config_formats.h"
+#endif
+
+#if (USE_PROTOBUF && USE_HDFS)
+
+#    include <google/protobuf/compiler/importer.h>
+#    include <google/protobuf/io/zero_copy_stream.h>
+#    include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+
+#    include <IO/HDFSCommon.h>
+#    include <IO/ReadBufferFromHDFS.h>
+
+namespace DB
+{
+class ProtobufHDFSSourceTree : public google::protobuf::compiler::SourceTree
+{
+public:
+    ProtobufHDFSSourceTree() { }
+    ~ProtobufHDFSSourceTree() override { }
+
+    void init(const std::string & hdfs_uri);
+    google::protobuf::io::ZeroCopyInputStream * Open(const std::string & filename) override;
+    std::string GetLastErrorMessage() override;
+
+private:
+    std::string last_error_message_;
+
+    HDFSBuilderPtr builder;
+    HDFSFSPtr fs;
+    std::string path;
+    std::string uri;
+};
+
+class ProtobufHDFSInputStream : public google::protobuf::io::ZeroCopyInputStream
```

Comment:
It seems this class isn't used anywhere except `ProtobufHDFSSourceTree.cpp`. I think it would be better to move it into an anonymous namespace in `ProtobufHDFSSourceTree.cpp` to make the public interfaces smaller.